src/rgw/rgw_op.cc

```diff
@@ -6239,7 +6239,9 @@ void RGWListMultipart::execute()
   mp.init(s->object.name, upload_id);
   meta_oid = mp.get_meta();
 
-  op_ret = get_multipart_info(store, s, meta_oid, &policy, nullptr, nullptr);
+  multipart_upload_info upload_info;
+  op_ret = get_multipart_info(store, s, meta_oid, &policy, nullptr, &upload_info);
+  storage_class = upload_info.dest_placement.storage_class;
```

Comment:
if you call `dest_placement.get_storage_class()` here instead of accessing it directly, you can avoid the call to `rgw_placement_rule::get_canonical_storage_class()` later

p.s. please move this assignment after the `if (op_ret < 0)` early return